src/index_runner/es_indexers/sample_set.py

```diff
@@ -112,46 +117,63 @@ def index_sample_set(obj_data, ws_info, obj_data_v1):
 
     for samp in data["samples"]:
         # query the sample service for sample
-        sample = _get_sample(samp)
-        sample_id = f"{_SAMPLE_NAMESPACE}::{sample['id']}:{sample['version']}"
-        # not sure on how we need to handle more than 1 node.
-        if len(sample['node_tree']) == 1:
-            meta_controlled = _flatten_meta(
-                sample['node_tree'][0]['meta_controlled']
-            )
-            meta_user = _flatten_meta(
-                sample['node_tree'][0]['meta_user']
-            )
-            meta_controlled['node_id'] = sample['node_tree'][0]['id']
+
+        sample_id = f"{_SAMPLE_NAMESPACE}::{samp['id']}"
+        if samp.get('version'):
+            sample_id += f":{samp['version']}"
+            sample = None
+        else:
+            # get latest version of sample
+            sample = _get_sample(samp)
+            sample_id += f":{sample['version']}"
+        # check if sample already indexed
+        document = _get_document("sample", sample_id)
+        if document:
+            # up date index to include this WS
+            document['sample_set_ids'].append(ver_sample_set_id)
         else:
-            meta_controlled, meta_user = {}, {}
-            for idx, node in enumerate(sample['node_tree']):
-                meta_controlled = _combine_meta(
-                    meta_controlled,
-                    _flatten_meta(
-                        node['meta_controlled']
-                    ),
-                    idx
+            if sample is None:
+                sample = _get_sample(samp)
+            # not sure on how we need to handle more than 1 node.
+            if len(sample['node_tree']) == 1:
+                meta_controlled = _flatten_meta(
+                    sample['node_tree'][0]['meta_controlled']
                 )
-                meta_user = _combine_meta(
-                    meta_user,
-                    _flatten_meta(
-                        node['meta_user']
-                    ),
-                    idx
+                meta_user = _flatten_meta(
+                    sample['node_tree'][0]['meta_user']
                 )
-                meta_controlled['node_id'] = node['id']
+                meta_controlled['node_id'] = sample['node_tree'][0]['id']
+            else:
+                meta_controlled, meta_user = {}, {}
+                for idx, node in enumerate(sample['node_tree']):
+                    meta_controlled = _combine_meta(
+                        meta_controlled,
+                        _flatten_meta(
+                            node['meta_controlled']
+                        ),
+                        idx
+                    )
+                    meta_user = _combine_meta(
+                        meta_user,
+                        _flatten_meta(
+                            node['meta_user']
+                        ),
+                        idx
+                    )
+                    meta_controlled['node_id'] = node['id']
```

Comment:
Are you sure you need conditional branching here? Isn't the case where you havemultiple node_tree entries the general case, and you could use that code for both branches?

Reply:
thats a good point. If I remember correctly elastic should flatten it either way